src/zenml/integrations/label_studio/annotators/label_studio_annotator.py

```diff
@@ -0,0 +1,696 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the Label Studio annotation integration."""
+
+import os
+import subprocess
+import sys
+import webbrowser
+from typing import Any, ClassVar, Dict, List, Optional, cast
+
+from label_studio_sdk import Client, Project  # type: ignore[import]
+
+from zenml.annotators.base_annotator import BaseAnnotator
+from zenml.exceptions import ProvisioningError
+from zenml.integrations.label_studio import LABEL_STUDIO_ANNOTATOR_FLAVOR
+from zenml.integrations.label_studio.steps.label_studio_standard_steps import (
+    LabelStudioDatasetRegistrationConfig,
+    LabelStudioDatasetSyncConfig,
+)
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.utils import io_utils, networking_utils
+
+logger = get_logger(__name__)
+
+DEFAULT_LABEL_STUDIO_PORT = 8093
+
+
+class LabelStudioAnnotator(BaseAnnotator):
```

Comment:
I forgot to mention this earlier: This component should also overwrite the `validator` property of the `StackComponent` base class and make sure it's only place in stacks that it can work with. That essentially means an azure, s3 or gcp artifact store I guess?

Reply:
Ok I think I added this correctly...

Reply:
Yes, looks good!